examples/cfd/darcy_deeponet_physics/physics_informed_deeponet.py

```diff
@@ -0,0 +1,327 @@
+# limitations under the License.
+
+import hydra
+from omegaconf import DictConfig
+import h5py
+import torch
+import numpy as np
+from sympy import Symbol, Function
+import matplotlib.pyplot as plt
+from hydra.utils import to_absolute_path
+import torch.nn.functional as F
+from torch.utils.data import Dataset, DataLoader
+
+from typing import Union
+from modulus.models.mlp import FullyConnected
+from modulus.models.fno import FNO
+from modulus.launch.logging import PythonLogger, LaunchLogger
+from modulus.launch.utils.checkpoint import save_checkpoint
+
+from modulus.sym.eq.pde import PDE
+
+
+class Darcy(PDE):
+    """Darcy PDE using Modulus Sym"""
+
+    name = "Darcy"
+
+    def __init__(self):
+
+        # time
+        x, y = Symbol("x"), Symbol("y")
+
+        # make input variables
+        input_variables = {"x": x, "y": y}
+
+        # make sol function
+        u = Function("sol")(*input_variables)
+        k = Function("K")(*input_variables)
+        f = 1.0
+
+        # set equation
+        self.equations = {}
+        self.equations["darcy"] = (
+            f
+            + k.diff(x) * u.diff(x)
+            + k * u.diff(x).diff(x)
+            + k.diff(y) * u.diff(y)
+            + k * u.diff(y).diff(y)
+        )
+
+
+class HDF5MapStyleDataset(Dataset):
+    """Simple map-style HDF5 dataset"""
+
+    def __init__(
+        self,
+        file_path,
+        device: Union[str, torch.device] = "cuda",
+    ):
+        self.file_path = file_path
+        with h5py.File(file_path, "r") as f:
+            self.keys = list(f.keys())
+
+        # Set up device, needed for pipeline
+        if isinstance(device, str):
+            device = torch.device(device)
+        # Need a index id if cuda
+        if device.type == "cuda" and device.index == None:
+            device = torch.device("cuda:0")
+        self.device = device
+
+    def __len__(self):
+        with h5py.File(self.file_path, "r") as f:
+            return len(f[self.keys[0]])
+
+    def __getitem__(self, idx):
+        data = {}
+        with h5py.File(self.file_path, "r") as f:
+            for key in self.keys:
+                data[key] = np.array(f[key][idx])
+
+        invar = torch.cat(
+            [
+                torch.from_numpy(
+                    (data["Kcoeff"][:, :240, :240] - 7.48360e00) / 4.49996e00
+                ),
+                torch.from_numpy(data["Kcoeff_x"][:, :240, :240]),
+                torch.from_numpy(data["Kcoeff_y"][:, :240, :240]),
+            ]
+        )
+        outvar = torch.from_numpy(
+            (data["sol"][:, :240, :240] - 5.74634e-03) / 3.88433e-03
+        )
+
+        x = np.linspace(0, 1, 240)
+        y = np.linspace(0, 1, 240)
+
+        xx, yy = np.meshgrid(x, y)
+        x_invar = torch.from_numpy(xx.astype(np.float32)).reshape(-1, 1)
+        y_invar = torch.from_numpy(yy.astype(np.float32)).reshape(-1, 1)
+
+        if self.device.type == "cuda":
```

Comment:
Can we get this from the distributed manager? Also, can we enable DDP for this example?